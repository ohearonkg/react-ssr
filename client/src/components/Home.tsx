import * as React from "react";

const Home = () => <div> I'm The Home Component </div>;
export default Home;
