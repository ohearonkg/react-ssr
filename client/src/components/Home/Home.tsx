import * as React from "react";

interface IHomeState {
  /**
   * Boolean representing
   * if some data has been
   * fetched
   */
  hasFetched: boolean
}
export default class Home extends React.Component<{},IHomeState> {

  constructor(props: null) {
    super(props);
    this.state = {
      hasFetched: false
    }
  }

  /**
   * Pretending to fetch something
   * from and API
   */
  public componentDidMount() {
    setTimeout(() => {
      this.setState({hasFetched: true})
    },3000)
  }

  public render() {
    const {hasFetched} = this.state;

    if (hasFetched === false) {
      return <h2> I am loading something </h2>
    }

    return (
      <div>I'm The Home Component</div>
    )
  }
}
