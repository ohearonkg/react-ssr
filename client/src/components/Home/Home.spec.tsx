import { shallow, ShallowWrapper } from "enzyme";
import toJson from "enzyme-to-json";
import * as React from "react";
import Home from "./Home";

describe("Home Component", () => {
  let wrapper: ShallowWrapper;
  beforeEach(() => (wrapper = shallow(<Home />)));

  /**
   * Snapshot
   */
  it("Should match its snapshot", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
