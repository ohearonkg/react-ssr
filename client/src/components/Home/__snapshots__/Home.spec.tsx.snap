// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component Should match its snapshot 1`] = `
<h2>
   I am loading something 
</h2>
`;
